src/org/scada_lts/web/mvc/api/MailingListAPI.java

```diff
@@ -75,4 +106,96 @@
+            } else {
+                return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+            }
+        } catch (Exception e) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @PostMapping(value = "", produces = "application/json")
+    public ResponseEntity<String> createMailingList(@RequestBody MailingList mailingList, HttpServletRequest request) {
+        LOG.info("POST:/api/mailingList");
+        try {
+            User user = Common.getUser(request);
+            if (user != null && user.isAdmin()) {
+                String error = validateMailingListCreate(mailingList);
+                if (!error.isEmpty()) {
+                    return ResponseEntity.badRequest().body(formatErrorsJson(error));
+                }
+                updateValueMailingListPost(mailingList);
+                mailingListService.saveMailingList(mailingList);
+                return new ResponseEntity<>("{\"status\":\"created\"}", HttpStatus.CREATED);
+            } else {
+                return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+            }
+        } catch (Exception e) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @PutMapping(value = "", produces = "application/json")
+    public ResponseEntity<String> updateMailingList(@RequestBody MailingListDTO mailingList, HttpServletRequest request) {
+        LOG.info("PUT:/api/mailingList");
+        try {
+            User user = Common.getUser(request);
+            if (user != null && user.isAdmin()) {
+                String error = validateMailingListUpdate(mailingList);
+                if (!error.isEmpty()) {
+                    return ResponseEntity.badRequest().body(formatErrorsJson(error));
+                }
+                return findAndUpdateMailingList(mailingList);
+            } else {
+                return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+            }
+        } catch (Exception e) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @DeleteMapping(value = "/{id}", produces = "application/json")
+    public ResponseEntity<String> deleteMailingListById(@PathVariable Integer id, HttpServletRequest request) {
+        LOG.info("DELETE: /api/mailingList/{id}");
+        try {
+            User user = Common.getUser(request);
+            if (user != null && user.isAdmin()) {
+                String error = validateMailingListDelete(id);
+                if (!error.isEmpty()) {
+                    return ResponseEntity.badRequest().body(formatErrorsJson(error));
+                }
+                mailingListService.deleteMailingList(id);
+                return new ResponseEntity<>("{\"status\":\"deleted\"}", HttpStatus.OK);
+            } else {
+                return new ResponseEntity<>(HttpStatus.UNAUTHORIZED);
+            }
+        } catch (Exception e) {
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
```

Comment:
Codes from group 400 should be reserved for validation errors, or if it does not find the object in the database (404), they refer to client errors, while the server error should be related to the code from group 500.